app/v2/controllers/actions/AcceptHeaderAction.scala

```diff
@@ -55,7 +56,13 @@ class AcceptHeaderActionImpl[R[_] <: Request[_]] @Inject() (implicit val executi
     }
 
   private def checkAcceptHeader[A](acceptHeaderValue: String, request: R[A]): Either[Result, R[A]] =
-    if (acceptedHeaders.contains(acceptHeaderValue)) Right(request)
-    else Left(NotAcceptable(Json.toJson(PresentationError.notAcceptableError(s"Accept header $acceptHeaderValue is not supported!"))))
+    if (
+      (!isOnlyJson && acceptedHeaders
+        .contains(acceptHeaderValue)) || (isOnlyJson && VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON.contains(acceptHeaderValue))
+    ) {
+      Right(request)
+    } else {
+      Left(NotAcceptable(Json.toJson(PresentationError.notAcceptableError(s"Accept header $acceptHeaderValue is not supported!"))))
```

Comment:
Can you change this message to `The Accept header is missing or invalid.` please, to bring it in line with comments on https://github.com/hmrc/common-transit-convention-traders/pull/386#pullrequestreview-1307583448?

Reply:
Yes updated